### PR TITLE
DataTable: Improvements to doDataTableSort

### PR DESCRIPTION
**Pull request by ElliotTheHuman**
### :sparkles: Changes
- Adds a variety of improvements to `doDefaultDataTableSort` including:
  - Now infers "string" if no type property is present on the corresponding column object
  - Supports default "date" sort
  - Is now a generic (and will set the type of the data property of the return object to match user's data interface)
- Docs + Storybook updates to showcase default "date" sort support
- JIRA Ticket: https://looker.atlassian.net/browse/LENS-1383

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [ ] Build and tests are passing
- [x] Update documentation
- [ ] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### CODE REVIEW

**Review by lukelooker — COMMENTED** [review]
Integrate into "interaction" stories and rename hook with "Default"

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `ComponentsProvider` now includes `FocusTrapContext` to manage all focus traps for `Dialog` and `Popover`
   - Where previously `DialogContext` properties `enableFocusTrap`, `disableFocusTrap`, and `focusTrapEnabled` could previously be used to take control of a focus trap, now use `FocusTrapContext` properties `enableCurrentTrap`, `disableCurrentTrap`, and `activeTrapRef` to do so.
   - `ScrollLockContext` properties keys have changed to match those on `FocusTrapContext`
+- `doDefaultDataTableSort` improvements
```
Comment (lukelooker) [inline packages/components/CHANGELOG.md:19]:
```suggestion
- `doDataTableSort` improvements (formerly `doDefaultDataTableSort`)
```

**Review by lukelooker — CHANGES_REQUESTED** [review]
Minor name change needed then it's GTG!

**Comment by ElliotTheHuman** [comment]
Brain farted and didn't push my local branch up before requesting the original review 🤦 

Resolved conflicts with master and pushed the name changes up.

**Review by lukelooker — APPROVED** [review]
LGTM :)

**Pull request merged**